src/network/protocol/messages/lib.zig

```diff
@@ -20,25 +22,29 @@ pub const Message = union(MessageTypes) {
     Verack: VerackMessage,
     Mempool: MempoolMessage,
     Getaddr: GetaddrMessage,
+    Block: BlockMessage,
     Getblocks: GetblocksMessage,
     Ping: PingMessage,
 
-    pub fn deinit(self: Message, allocator: std.mem.Allocator) void {
-        switch (self) {
+    pub fn deinit(self: *Message, allocator: std.mem.Allocator) void {
+        switch (self.*) {
             .Version => |m| m.deinit(allocator),
             .Verack => {},
             .Mempool => {},
             .Getaddr => {},
+            .Block => |*m| m.deinit(allocator),
             .Getblocks => |m| m.deinit(allocator),
```

Comment:
can you also change it for the others stuct types pliz?

Reply:
Other struct don't need this, once I'll have changed ArrayList for slice, it will not be required anymore

Reply:
Still, it is often better to pass a struct by reference rather than value